analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RegularExpressions/RegexMustHaveValidSyntax.CSharp9.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Text.RegularExpressions;
+
+class Compliant
+{
+    void ImplicitObject()
+    {
+        Regex defaultOrder = new("valid pattern"); // Compliant
+    }
+}
+
+class Noncompliant
+{
+    private const string InvalidPattern = "[A";
+
+    void ImplicitObject()
+    {
+        Regex patternOnly = new("[A"); // Noncompliant {{Fix the syntax error inside this regex: Invalid pattern '[A' at offset 2. Unterminated [] set.}}
+        Regex differentIssue = new("A???"); // Noncompliant {{Fix the syntax error inside this regex: Invalid pattern 'A???' at offset 4. Nested quantifier '?'.}}
```

Comment:
Aligning them to the same tab distance will make it way easier to see what's the common part and what has changed
```
        Regex patternOnly = new("[A");      // Noncompliant {{Fix the syntax error inside this regex: Invalid pattern '[A' at offset 2. Unterminated [] set.}}
        Regex differentIssue = new("A???"); // Noncompliant {{Fix the syntax error inside this regex: Invalid pattern 'A???' at offset 4. Nested quantifier '?'.}}
```